Source/AMRInterpolator/SurfaceGeometry.hpp

```diff
@@ -0,0 +1,64 @@
+/* GRChombo
+ * Copyright 2012 The GRChombo collaboration.
+ * Please refer to LICENSE in GRChombo's root directory.
+ */
+
+#ifndef SURFACEGEOMETRY_HPP_
+#define SURFACEGEOMETRY_HPP_
+
+#include <string>
+
+//! An abstract base class to provide required methods for geometries in the
+//! SurfaceExtraction class
+class SurfaceGeometry
+{
+  public:
+    //! returns the grid spacing in u
+    virtual double du(int a_num_points_u) const = 0;
```

Comment:
Most of the things in SphericalGeometry.hpp should have inline's, right?

Reply:
Unfortunately you can't inline pure virtual functions (as far as I'm aware). This goes to back to one of my questions in the original post about whether this generalisation using pointers to abstract base classes should just be changed to templating over a `SurfaceGeometry` class instead?